herbie_core/services/schema_mock_data/schema_string_generator.py

```diff
@@ -0,0 +1,43 @@
+from faker import Faker
+import rstr
+
+from herbie_core.constants import JsonSchemaPropertiesConstants
+
+
+class SchemaStringGenerator:
+    def __init__(self):
+        self._faker = Faker()
+
+    def generate_string(self, schema):
+        format = schema.get(JsonSchemaPropertiesConstants.STRING_TYPE_FORMAT, None)
+        min_length = schema.get(JsonSchemaPropertiesConstants.STRING_TYPE_MIN_LENGTH, None)
+        max_length = schema.get(JsonSchemaPropertiesConstants.STRING_TYPE_MAX_LENGTH, None)
+        enum = schema.get(JsonSchemaPropertiesConstants.STRING_TYPE_ENUM, None)
+        pattern = schema.get(JsonSchemaPropertiesConstants.STRING_TYPE_PATTERN, None)
+
+        if pattern is not None:
+            return rstr.xeger(pattern)
```

Comment:
Question: Can it be a pattern and an enum at the same time?

Reply:
Don't think so, it's specifically just for regex:
https://json-schema.org/understanding-json-schema/reference/string.html#regular-expressions

Reply:
:+1: